renku/core/metadata/repository.py

```diff
@@ -423,6 +425,65 @@ def get_content_from_submodules():
 
         raise errors.ExportError(f"File not found in the repository: '{revision}/{checksum}:{path}'")
 
+    def get_object_hashes(self, paths: List[Union[Path, str]], revision: str = None) -> Dict[str, str]:
```

Comment:
The returned dictionary from this function does not have consistent keys: For modified objects the key is an absolute path while for the rest it's a relative path.
I believe it should return return the passed input values as keys so that we can reliably get a hash value from it (converting Path to str is ok). Ideally, if a path doesn't exists in the repo then it's hash should be set to None in the returned dict.

Reply:
This function is a bit difficult to follow. Can you please use nested/inner functions to make it more readable?